inc/Engine/Cache/Purge.php

```diff
@@ -136,4 +151,73 @@ private function maybe_remove_dir( $dir ) {
 			rocket_rrmdir( $dir, [], $this->filesystem );
 		}
 	}
+
+	/**
+	 * Purge all terms archives urls associated to a specific post.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param WP_Post $post Post object.
+	 */
+	public function purge_post_terms_urls( WP_Post $post ) {
+		foreach ( $this->get_post_terms_urls( $post ) as $url ) {
+			$this->purge_url( $url );
+		}
+
+	}
+
+	/**
+	 * Get all terms archives urls associated to a specific post.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param WP_Post $post Post object.
+	 *
+	 * @return array $urls    List of taxonomies URLs
+	 */
+	private function get_post_terms_urls( WP_Post $post ) {
+		$urls       = [];
+		$taxonomies = get_object_taxonomies( get_post_type( $post->ID ), 'objects' );
+
+		foreach ( $taxonomies as $taxonomy ) {
+			if ( ! $taxonomy->public || 'product_shipping_class' === $taxonomy->name ) {
```

Comment:
@Tabrisrp any ideas if this is a good option and if we should do it?

Reply:
@arunbasillal this is a minor change, we can ignore it for now so u can proceed on QA review, thanks

Reply:
I think it's a very good idea to use a filter here and populate it based on third party needs

Reply:
@Tabrisrp & @engahmeds3ed we should create a separate PR for this one so that @arunbasillal can test it. Should we create also a new github issue for it?

Reply:
I can test it in the same PR for sure. But if its worth adding to the changelog separately, having a separate issue/PR would be useful.

Reply:
@crystinutzaa if you have time for this sprint to create a separate PR, go ahead. Else we can create an issue for it and work on it later